### ISSUE TITLE
SyntaxError: Expected "," or '}' after property value in JSON at position 336

### ISSUE THREAD

**Issue by tbernacchi** [rancher/dashboard#7285]
**Setup**
- Rancher version: 2.6.6
- Browser type & version:

**Describe the bug**

When trying to import or create an Azure/AKS cluster with client secret keys expired, the UI throws: 
![image](https://user-images.githubusercontent.com/26934246/197764900-9be6a571-8f29-45bd-9592-9df7055f9519.png)

**To Reproduce**

- Try to import or create an Azure/AKS cluster with client secret keys expired.

**Result**
![image](https://user-images.githubusercontent.com/26934246/197765075-a5a274c3-9a18-411e-ae1e-10aee4b5a15a.png)

**Expected Result**

A more friendly error message according to what the logs of Rancher it's saying:

```
The provided client secret keys for app 'a59c8913-d4df-4752-ae69-f6488ab00d62' are expired. Visit the Azure portal to create new keys for your app
```

**Screenshots**

<img width="1915" alt="secrets-expired-azure" src="https://user-images.githubusercontent.com/26934246/197766685-4bbc84c4-7b9e-4111-a70b-20e2f834cea2.png">

**Additional context**

**Comment by snasovich** [comment]
The initial root cause of this issue is invalid JSON `aksLocations` endpoint produces when error message contains `"`s. This is being fixed in https://github.com/rancher/rancher/pull/39612 for the next 2.7.x release. Please note the following from the PR details:
> Following this change, in the situation outlined in https://github.com/rancher/dashboard/issues/7285 UI now shows `[object Object]` error text that is not helpful - but not less helpful than the current `SyntaxError: Expected "," or '}' after property value in JSON at position...`. UI code needs to be improved to properly handle the returned error object.

**Comment by cpinjani** [comment]
Verified as fixed on build: `v2.9-360c02b3761f7f2bf621cde5767eb65b67dcc059-head`

When trying to create an AKS cluster with client secret keys expired, the UI throws proper error which matches with backend logs:
![image](https://github.com/rancher/dashboard/assets/73099870/a98b8fa5-4e6f-4dac-907d-88b4f95ef22a)

**Logs:**
2024/01/09 05:42:31 [ERROR] [aks-handler] error getting locations: failed to get locations: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscriptions/REDACTED/locations?api-version=2016-06-01: StatusCode=401 -- Original Error: adal: Refresh request failed. Status Code = '401'. Response body: {"error":"invalid_client","error_description":"AADSTS7000222: The provided client secret keys for app 'de9ef773-cdcd-4ce8-bfca-6982d555f417' are expired. Visit the Azure portal to create new keys for your app: https://aka.ms/NewClientSecret, or consider using certificate credentials for added security: https://aka.ms/certCreds. Trace ID: c1fde356-7447-41d2-97cf-4e8ffe453400 Correlation ID: 9998d949-6440-451c-a6ff-09515328ce8b Timestamp: 2024-01-09 05:42:31Z","error_codes":[7000222],"timestamp":"2024-01-09 05:42:31Z","trace_id":"c1fde356-7447-41d2-97cf-4e8ffe453400","correlation_id":"9998d949-6440-451c-a6ff-09515328ce8b","error_uri":"https://login.microsoftonline.com/error?code=7000222"} Endpoint https://login.microsoftonline.com/fcf34994-aac4-4462-afaa-d83f87c5f51d/oauth2/token?api-version=1.0
